### ISSUE TITLE
`fix` sometimes prints error message to stdout instead of stderr

### ISSUE THREAD

**Issue by syphar** [sqlfluff/sqlfluff#3892]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I was digging into https://github.com/jose-elias-alvarez/null-ls.nvim/issues/1123 and found out that in the described case sqlfluff does output one of the error messages not to stderr but to stdout. 

In case of null-ls this will lead to the output being inserted into the buffer. 

### Expected Behaviour

error messages are only print to `stderr`

### Observed Behaviour

see "how to reproduce"

### How to reproduce

using this SQL file: 
```sql
CREATE TABLE IF NOT EXISTS vuln.software_name_dictionary(
    id SERIAL PRIMARY KEY,
    rule VARCHAR(30),
);
```

The output of sqlfluff for this file is: 
```
$ cat test.sql | sqlfluff fix --disable_progress_bar -f -n --dialect postgres -   
WARNING    One fix for L052 not applied, it would re-cause the same error.
WARNING    One fix for L052 not applied, it would re-cause the same error.
WARNING    One fix for L052 not applied, it would re-cause the same error.
WARNING    One fix for L052 not applied, it would re-cause the same error.
WARNING    One fix for L052 not applied, it would re-cause the same error.
WARNING    One fix for L052 not applied, it would re-cause the same error.
WARNING    One fix for L052 not applied, it would re-cause the same error.
WARNING    One fix for L052 not applied, it would re-cause the same error.
WARNING    One fix for L052 not applied, it would re-cause the same error.
WARNING    One fix for L052 not applied, it would re-cause the same error.
WARNING    Loop limit on fixes reached [10].
  [1 templating/parsing errors found]
Unfixable violations detected.
CREATE TABLE IF NOT EXISTS vuln.software_name_dictionary(
    id SERIAL PRIMARY KEY,
    rule VARCHAR(30),
);
```

but redirecting stderr I still see one of the messages: 
```
$ cat test.sql | sqlfluff fix --disable_progress_bar -f -n --dialect postgres - 2>/dev/null
  [1 templating/parsing errors found]
CREATE TABLE IF NOT EXISTS vuln.software_name_dictionary(
    id SERIAL PRIMARY KEY,
    rule VARCHAR(30),
);
```

I would expect the error message (_  [1 templating/parsing errors found]_) in stderr, not in stdout. 

### Dialect

postgresql

### Version

- sqlfluff, version 1.3.2
- python 3.10.6

### Configuration

none

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by syphar** [comment]
With some hints where to look I'm happy to work on a PR.

**Comment by KingMichaelPark** [comment]
https://github.com/sqlfluff/sqlfluff/compare/main...KingMichaelPark:sqlfluff:main

Worked on this, will run all the unit tests tomorrow but this section should cover it, will test against the output as well to triple check all is okay. Just letting you know now @syphar  so you don't waste time on it

**Comment by KingMichaelPark** [comment]
![image](https://user-images.githubusercontent.com/64011084/193401154-46e37077-2348-45ac-af60-c0436bd192b4.png)

https://github.com/sqlfluff/sqlfluff/pull/3900
